controllers/release/adapter.go

```diff
@@ -69,6 +69,39 @@ func newAdapter(ctx context.Context, client client.Client, release *v1alpha1.Rel
 	}
 }
 
+// EnsureAuthorIsValid is an operation that will ensure that Release is attributed to a specific,
+// valid user.
+func (a *adapter) EnsureAuthorIsValid() (controller.OperationResult, error) {
+	if !a.release.IsReleasing() {
+		return controller.ContinueProcessing()
+	}
+
+	if a.release.Labels[metadata.AutomatedLabel] == "true" && !a.release.IsAutomated() {
+		return controller.RequeueWithError(fmt.Errorf("automated not set in status for automated release"))
+	}
+
+	patch := client.MergeFrom(a.release.DeepCopy())
+	releasePlan, err := a.loader.GetReleasePlan(a.ctx, a.client, a.release)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.release.MarkValidationFailed(err.Error())
+			a.release.MarkReleaseFailed("Release validation failed")
+			return controller.RequeueOnErrorOrStop(a.client.Status().Patch(a.ctx, a.release, patch))
+		}
+
+		return controller.RequeueWithError(err)
+	}
+
+	err = a.registerAttributionData(releasePlan)
```

Comment:
Once again you have a patch within a patch :stuck_out_tongue_winking_eye:

Reply:
fixed. removed patch from registerAttributionData

Reply:
As we said offline, `register` functions should actually use patch.